### ISSUE TITLE
Send notification when reservation is being sent

### ISSUE THREAD

**Issue by revi** [enwikipedia-acc/waca#487]
The target user (who receives it) should have a more reliable way of knowing they’ve been assigned, other than not-that-reliable irc logging.

Options could be 1. Sending an email / 2. Make some notification tab or such on interface and make them display notifications.

**Comment by stwalkerster** [comment]
Email notifications could probably be done in oldinternal, though there would be no way of opting out of these. It's verging on the larger of the things I'd want to add to oldinternal at this point though (due to difficulties merging it up to newinternal later)

A notifications tab thing I like the idea of, but that's definitely in newinternal territory as it's probably going to need quite some effort to deal with.

I'd like @FastLizard4 's opinion on this too, since anything we do potentially impacts newinternal's timescales.

**Comment by ElHefWiki** [comment]
+1 support this. I received a couple of reservations today in a queue I don't normally check, and the only reason I knew I had them was the warning that I had multiple requests reserved when I handled a normal request, and went looking for why.

**Comment by stwalkerster** [comment]
Right, since we seem to be using the send reservation feature more and more, I think this is something we need to prioritise.

From a functional design standpoint, I think I want to do something like a stripped-down version of MediaWiki's Echo. Prior warning, this is very high-level and very... vague.

I would like certain events to trigger "notifications":
* `@username` in the comments field
* Sending a reservation to someone

Notifications will be shown to the user on the next page load, probably as banner notifications (bootstrap "alerts") until dismissed, pending a better UI. I don't expect notifications to be particularly frequent, so this is probably sufficient for a first iteration.

A new user preference should be added as a "CC" option on the alerts, to request a user log in. Alerts will be shown in the UI regardless. Again, first iteration UI, if we want to extend it later we can.

I'll also add a dependency on #44 here - something we *really* should have added years ago, but haven't yet. At the moment, email is pretty much only used for password reset and account status change notification IIRC.

Does this sound like a reasonable first iteration?
